pkg/cloud/services/compute/instance.go

```diff
@@ -562,38 +561,22 @@ func applyServerGroupID(opts servers.CreateOptsBuilder, serverGroupID string) se
 	return opts
 }
 
-// Helper function for getting image id from name.
-func (s *Service) getImageIDFromName(imageName string) (string, error) {
-	var opts images.ListOpts
-
-	opts.Name = imageName
-
-	allImages, err := s.getImageClient().ListImages(opts)
+// Helper function for getting image ID from name or ID.
+func (s *Service) getImageID(image infrav1.ImageFilter) (string, error) {
+	allImages, err := s.getImageClient().ListImages(image.ToListOpt())
 	if err != nil {
 		return "", err
 	}
 
 	switch len(allImages) {
 	case 0:
-		return "", fmt.Errorf("no image with the Name %s could be found", imageName)
+		return "", fmt.Errorf("no images could be found")
```

Comment:
should we add the filter info into list?

Reply:
good idea, i'm doing this.